### PR TITLE
feat: add types for BigCommerce SDK

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
Add files with types for BigCommerce SDK

## Description
Types moved to _api-client/types_ directory. Create helpers directory with helper for BigCommerce client

## Motivation and Context
Prepares project for the BigCommerce integration

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/api-client/src/types/context/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { IntegrationContext } from '@vue-storefront/core';
+import { NodeBigcommerceSDK } from '../nodeBigcommerceSDK';
+import { Endpoints } from '../api/endpoints';
+
```
Comment (scernisevs) [inline packages/api-client/src/types/context/index.ts:4]: wanted to ask if a doc comment would also add value for `MiddlewareSDKSettingsConfig`?

File: packages/api-client/src/types/nodeBigcommerceSDK/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+type REQUEST_TYPE = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+
+/**
+ * The definition of types for SDK provided by Bigcommrce team
```
Comment (scernisevs) [inline packages/api-client/src/types/nodeBigcommerceSDK/index.ts:4]: a minor typo, 'Bigcommrce' -> 'Bigcommerce'

**Review by scernisevs — COMMENTED** [review]
1 very minor typo in the comment block and 1 question from my side, everything else looks good!

File: packages/api-client/src/types/context/index.ts (hunk 1 above)
Comment (WojtekTheWebDev) [reply packages/api-client/src/types/context/index.ts:4]: Missed that, thanks, fixed now

File: packages/api-client/src/types/nodeBigcommerceSDK/index.ts (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/api-client/src/types/nodeBigcommerceSDK/index.ts]: Thanks, fixed

**Review by scernisevs — APPROVED** [review]

**Pull request merged**